File: frontend/src/Services/PatientService.js
import axios from "axios";

class PatientService{

    getAppointments(email){
        return axios.get("http://localhost:8080/api/getPatientAppointments/"+email);
    }

    getPatientProfile(email){
        return axios.get("http://localhost:8080/api/getPatientProfile/"+email);
    }

    updatePatient(patient){
        return axios.put(`http://localhost:8080/api/updatePatientProfile/${patient.patientId}`,null, {
            params: {
            name: patient.name,
            phoneNumber:patient.phoneNumber,
            age: patient.age,
            gender: patient.gender
          }});
    }

    changePassword(password,email){

        return axios.post(`http://localhost:8080/api/patient_changepassword/${email}`, null, {
    params: {
    currpassword: password.currpassword,
    newpassword: password.newpassword,
    confirmpassword: password.confirmpassword
  }
});
}


}

export default new PatientService();

File: backend/src/main/java/com/appointment/Service/PatientService.java
package com.appointment.Service;

import java.util.List;

import com.appointment.Entity.Appointment;
import com.appointment.Entity.Patient;



public interface PatientService {

    public Patient AddPatient(String name,String email,String password);
	
	public boolean deletePatient(int id);
	
	public Patient getPatientById(int id);
	
	public Patient updatePatient(int id,String name,String phonenumber,int age,String gender);
	
	public List<Patient> getAllPatients();
	
	public List<Appointment> getAppointments(String email);
	
	public Patient getPatientByUsername(String username);
	
}
